src/main/helm/helm-release-manager.ts

```diff
@@ -22,161 +22,224 @@
 import * as tempy from "tempy";
 import fse from "fs-extra";
 import * as yaml from "js-yaml";
-import { promiseExec } from "../../common/utils/promise-exec";
+import { promiseExecFile } from "../../common/utils/promise-exec";
 import { helmCli } from "./helm-cli";
-import type { Cluster } from "../cluster";
 import { toCamelCase } from "../../common/utils/camelCase";
+import type { BaseEncodingOptions } from "fs";
+import { execFile, ExecFileOptions } from "child_process";
 
-export async function listReleases(pathToKubeconfig: string, namespace?: string) {
-  const helm = await helmCli.binaryPath();
-  const namespaceFlag = namespace ? `-n ${namespace}` : "--all-namespaces";
+async function execHelm(args: string[], options?: BaseEncodingOptions & ExecFileOptions): Promise<string> {
+  const helmCliPath = await helmCli.binaryPath();
 
   try {
-    const { stdout } = await promiseExec(`"${helm}" ls --output json ${namespaceFlag} --kubeconfig ${pathToKubeconfig}`);
-    const output = JSON.parse(stdout);
-
-    if (output.length == 0) {
-      return output;
-    }
-    output.forEach((release: any, index: number) => {
-      output[index] = toCamelCase(release);
-    });
+    const { stdout } = await promiseExecFile(helmCliPath, args, options);
 
-    return output;
+    return stdout;
   } catch (error) {
     throw error?.stderr || error;
```

Comment:
I know this is unrelated but shouldn't it just be
```suggestion
    throw error.stderr || error;
```
?

Reply:
I guess so. But since any value can be thrown I sort of did it to guard against null and undefined values. But I guess it doesn't guard against other non object types.

Reply:
Then there's no point having the `||`. This gives same result? Or what am I missing?
```suggestion
    throw error?.stderr;
```

Reply:
Well if it is an error instance that doesn't have `stderr` we still want the error.

Reply:
`error` from `catch(error)` can be undefined?

Reply:
yes, `throw;` is valid JS/TS